src/main/java/com/microsoft/sqlserver/jdbc/IOBuffer.java

```diff
@@ -1895,88 +1626,109 @@ else if (con.getTrustManagerClass() != null) {
             // Otherwise, we'll validate the certificate using a real TrustManager obtained
             // from the a security provider that is capable of validating X.509 certificates.
             else {
-                if (logger.isLoggable(Level.FINER))
-                    logger.finer(toString() + " SSL handshake will validate server certificate");
+                if (isTDSS) {
+                    if (logger.isLoggable(Level.FINEST))
+                        logger.finest(toString() + " Verify server certificate for TDSS");
+
+                    if (logger.isLoggable(Level.FINEST))
```

Comment:
1630, 1633 and 1638 are all checking the same condition. All this logging can be moved under the first FINEST check.